src/app/services/transmart-services/transmart-external-job-resource.service.ts

```diff
@@ -0,0 +1,260 @@
+          ErrorHelper.handleError(error);
+          return observableOf(error);
+        })
+      );
+    }
+  }
+
+  /**
+   * Get the current user's existing export jobs
+   * @returns {Observable<any[]>}
+   */
+  getAllJobs(): Observable<TransmartExternalJob[]> {
+    const urlPart = 'jobs';
+    const responseField = 'jobs';
+    return this.getCall(urlPart, responseField);
+  }
+
+  /**
+   * Get a status of the job
+   * @param jobId
+   * @returns {Observable<ExportJob>}
+   */
+  getJobStatus(jobId: string): Observable<TransmartExternalJob> {
+    const urlPart = `jobs/status/${jobId}`;
+    const responseField = 'exportJob';
+    return this.getCall(urlPart, responseField);
+  }
+
+  /**
+   * Create and run an export job for the current user, with a given name
+   *
+   * @param {string} jobName
+   * @param {Constraint} constraint
+   * @returns {Observable<ExportJob>}
+   */
+  runJob(jobName: string, constraint: Constraint): Observable<TransmartExternalJob> {
+    const urlPart = `jobs/create`;
+    const responseField = 'job';
+    let targetConstraint = constraint;
+    if (this.autosaveSubjectSets &&
+      constraint.className === 'CombinationConstraint' &&
+      (<CombinationConstraint>constraint).children[1].mark === ConstraintMark.OBSERVATION) {
+      let combo = new CombinationConstraint();
+      combo.addChild(this.subjectSetConstraint);
+      combo.addChild((<CombinationConstraint>constraint).children[1]);
+      targetConstraint = combo;
+    }
+
+    let body = {
+      job_type: this.customExportJobName,
+      job_parameters: {
+        constraint: TransmartConstraintMapper.mapConstraint(targetConstraint),
+        custom_name: jobName
+      }
+    };
+
+    return this.postCall(urlPart, body, responseField);
+  }
+
+  /**
+   * Given an export job id, return the blob (zipped file) ready to be used on frontend
+   * @param jobId
+   * @returns {Observable<blob>}
+   */
+  downloadJobData(jobId: string) {
+    let url = `${this.endpointUrl}/jobs/data/${jobId}`;
+    return this.http.get(url, {responseType: 'blob'}).pipe(
+      catchError(ErrorHelper.handleError.bind(this)));
+  }
+
+  /**
+   * Cancels an export job with the given export job id
+   * @param jobId
+   * @returns {Observable<blob>}
+   */
+  cancelJob(jobId: string): Observable<{}> {
+    const urlPart = `jobs/cancel/${jobId}`;
+    return this.getCall(urlPart, null);
+  }
+
+
+  // -------------------------------------- mocked calls --------------------------------------
+
+  /**
+   * Creates mocked export job,
+   * when the external job handler has one method to both create and run the job
+   * @param {string} name
+   * @returns {Observable<ExportJob>}
+   */
+  createExportJob(name: string): Observable<ExportJob> {
+    let newExportJob = new ExportJob();
+    newExportJob.id = name;
+    newExportJob.jobName = name;
+    return observableOf(newExportJob);
+  }
+
+  /**
+   * Returns mocked file formats,
```

Comment:
Why 'mocked', these are the actual supported file formats, right?

Reply:
true, I meant mocked call comparing to the transmart-resource-service, maybe at some point there will be such a call in tm-packer. Will change this comment.